test/ZendeskApi.Client.Tests/Resources/JobStatusResourceTests.cs

```diff
@@ -0,0 +1,125 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging.Abstractions;
+using Xunit;
+using ZendeskApi.Client.Exceptions;
+using ZendeskApi.Client.Models;
+using ZendeskApi.Client.Resources;
+using ZendeskApi.Client.Tests.ResourcesSampleSites;
+
+namespace ZendeskApi.Client.Tests.Resources
+{
+    public class JobStatusResourceTests
+    {
+        private readonly JobStatusResource _resource;
+
+        public JobStatusResourceTests()
+        {
+            IZendeskApiClient client = new DisposableZendeskApiClient<JobStatusResponse>(resource => new JobStatusSampleSite(resource));
+            _resource = new JobStatusResource(client, NullLogger.Instance);
+        }
+
+        [Fact]
+        public async Task ListAsync_WhenCalled_ShouldGetAll()
+        {
+            var results = await _resource.ListAsync();
+
+            Assert.Equal(100, results.Count);
+
+            for (var i = 1; i <= 100; i++)
```

Comment:
We have this same loop in almost all the new test classes. Can we pull it out into it's own method we can reuse?

Reply:
Any ideas on how? The model we assert against is different every time. You could create a func to pass in the assertions but it kinda feels a bit pointless just hiding a for loop?

Reply:
Oh yeah, that's true. Probably not much point then.